README.md

```diff
@@ -1,15 +1,101 @@
+
+> To see a step-by-step example of how to install this action into your software project, [review the official AppMap Documentation](http://appmap.io/docs/analysis/in-github-actions).
+
 GitHub action to build and store an AppMap archive.
 
-To see a step-by-step example of how to install this action into your software project, [review the official AppMap Documentation](http://appmap.io/docs/analysis/in-github-actions).
+## Table of contents
+
+- [Prerequisites](#prerequisites)
+- [Inputs](#inputs)
+- [Examples](#examples)
+- [Development](#development)
+
+## Prerequisites
+
+The `archive-action` needs to run AFTER your test cases have executed to build and save an AppMap archive to the GitHub build asset repository. Your application will need AppMap libraries and configuration files created in the project, or you will need to use the [`install-action`](https://github.com/getappmap/install-action) to setup AppMap before your test cases run. 
+
+For more information about how to setup AppMap in your project [review the official AppMap Documentation](http://appmap.io/docs/analysis/in-github-actions).
+
+## Inputs
+
+Add a step like this to your workflow:
+
+```yaml
+- name: Archive AppMaps
+  uses: getappmap/archive-action@v1
+  with:
+    # Command working directory. Change this if your project lives in a 
+    # subdirectory or for monorepo / multi-project support
+    # Default: '.'
+    directory: /path/to/code
+
+    # 0-based numerical id of a matrix job. This id is used to store the AppMap 
+    # archive in the action cache. Once the matrix jobs complete, the `merge` 
+    # action can be used to merge the archives into a single archive, and then 
+    # save it as an artifact. If `archive-id` is set, the archive is saved to the 
+    # cache. Otherwise, it's uploaded to the artifact store. For a non-matrix 
+    # job, `archive-id` should not be used.
+    # Required for Multi-runner matrix builds.
+    archive-id: ${{ matrix.ci_node_index }}
+
+    # Revision (commit SHA) to name the AppMap archive with. The default is the 
+    # GITHUB_SHA. Under normal circumstances, the default value is correct 
+    # and this input does not need to be specified.
+    revision: 6056b0cfbacd562f4a4f274122c4a5e85542e040
+      
+    # The GitHub token to use with the GitHub API to enumerate AppMap Tools 
+    # releases.
+    # Default: `${{ github.token }}`
+    github-token: secrets.CUSTOM_GITHUB_TOKEN
+    
+    # Number of worker threads to use for processing the archive. Defaults to the 
+    # number of CPUs / cores, as reported by Node.js. If the worker machine has 
+    # a high number of CPUs/cores, the archive action may become I/O-bound rather 
+    # than CPU-bound, and better performance might be obtained by setting this 
+    # value to a lower number.
+    thread-count: 4
+        
+    # Enable verbose logging.
+    # Default: false
+    verbose: true
+```
+
+## Examples
```

Comment:
Looking over this it seems like it would be helpful to split the examples by scenario:

1) Initial setup of a project
2) Code review - pull_request trigger - archive a matrix worker + merge the worker data
3) Merge of a PR - non-matrix - push or schedule trigger - archive the new baseline AppMaps
4) Merge of a PR - matrix - push or schedule trigger - collect and archive the AppMaps

Reply:
actually the non matrix won't even use this - we use the cache save then send to the reusable workflow.  It's just for matrix builds.

Reply:
That's true but since we are documenting the action, we don't assume that the user using the reusable workflow. This shows them how it actually works in case they want to roll their own.